libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SoqlSyncDownTarget.java

```diff
@@ -183,7 +202,15 @@ public static String addFilterForReSync(String query, String modificationFieldDa
     /**
      * @return soql query for this target
      */
-	public String getQuery() {
-        return query;
+    public String getQuery() {
+        return getQuery(0);
+    }
+
+    /**
+     * @return soql query for this target
+     * @param maxTimeStamp
+     */
+	public String getQuery(long maxTimeStamp) {
```

Comment:
The subclass is building the SOQL query and so does need to rely on some nasty regexp replace to inject the time stamp predicate